posts/2023-10-20-open-liberty-with-cargo-tracker-ee10.adoc

```diff
@@ -0,0 +1,119 @@
+
+image::/img/blog/cargo-tracker-ee10-home.png[Cargo Tracker Home Page,width=50%,align="left"]
+image::/img/blog/cargo-tracker-ee10-map.png[Cargo Tracker Map,width=50%,align="right"]
+
+
+== Run Cargo Tracker with Open Liberty 
+
+Cargo Tracker EE10 supports Java SE 11 and 17. https://developer.ibm.com/languages/java/semeru-runtimes/downloads[IBM Semeru Runtime] is the preferred Java runtime when using Open Liberty. Before you clone the application to your machine, install any required JDKs and ensure that your `JAVA_HOME` system variable is set to a compatible version. To access the repository remotely you'll need to install https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git] if you haven't already. Finally, you will need to install https://maven.apache.org/download.cgi[Maven] to build and run the application. 
+
+You can clone the Cargo Tracker project to your local machine by running:
+
+[source]
+----
+git clone https://github.com/eclipse-ee4j/cargotracker.git
+----
+
+You can also visit the https://github.com/eclipse-ee4j/cargotracker/tree/master[Git repository the Cargo Tracker project] to take a peek at the code.
+
+To get started, choose one of the following setup methods:
+
+* <<maven_setup, Maven command line setup>> 
+* <<vscode_setup, Visual Studio Code IDE setup>>
+
+https://openliberty.io/docs/latest/development-mode.html[Liberty dev mode] is used in either setup because it provides three key features. Automatically detect, recompile, and deploy code changes in your running Liberty. Unit and integration tests are run on demand when you press `Enter` in the command terminal. Finally, you can attach a debugger to the running Liberty to step through the code. Through those features, you can deeply learn the Jarakee EE through debugging or updating the Cargo Tracker code.
+
+
+[#maven_setup]
+== Maven command line setup
+
+After cloning the project, you can quickly start Cargo Tracker through the command line with Maven. Navigate to the project base directory (where the pom.xml file is located) and run:
+
+[source]
+----
+mvn -Popenliberty liberty:dev
+----
+ 
+The application should start without any thrown exceptions. You can safely ignore the shrinkwrap features warning, message-driven bean warning, and the AggregateObjectMapping nested foreign key warning, as they don't affect the application functionality.
+
+image::/img/blog/cargo-tracker-ee10-dev-mode.png[Cargo Tracker Maven Output,width=70%,align="center"]
+
+
+[#vscode_setup]
+== Visual Studio Code IDE setup
+
+Ensure you are running Visual Studio Code (VSCode) IDE version 1.78.0 or later with Java 17. It is recommended to install Liberty Tools on your IDE from the https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[marketplace].
```

Comment:
Can make this a little less formal. For the second sentence: "We'd also recommend installing the Liberty Tools for Visual Studio Code extension in your IDE to benefit from the fast and iterative development features it offers directly within the IDE. You can download this extension directly from the VS Code marketplace."

Reply:
Should maybe also add that this extension is used in this demo/these screenshots/instructions as otherwise they won't have the dashboard used for the instructions after this.

Reply:
> Should maybe also add that this extension is used in this demo/these screenshots/instructions as otherwise they won't have the dashboard used for the instructions after this.

No sure what it means. Are you talking for the Liberty Tools doc to contain the screenshots?